cpp/ql/test/experimental/query-tests/Security/CWE/CWE-119/test.cpp

```diff
@@ -243,3 +243,15 @@ void test_flow_through_setter(unsigned size) {
     memset(str.string, 0, size + 1); // BAD
 }
 
+unsigned write(int, const void *, unsigned);
+
+void test_modify_pointer_and_size_in_sync(int sock, unsigned size) {
+  int *p = new int[size];
+  while (size > 0) {
+    unsigned n = write(sock, p, size); // GOOD [FALSE POSITIVE]
+    if (n > 0) {
+      size += 1;
+      p -= 1;
```

Comment:
This seems incorrect. This updates `p` with a value that's definitely not pointing to the allocated `int` array.

Reply:
Oops, you're right. It was supposed to say `size -= 1; p += 1`, but when I do that the FP is gone 😭. I'll investigate!

Reply:
Hm. Turns out the flow path is actually a whole lot more complicated than I had first realized. And there may not be a problem with the query after all 🎉